packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -72,6 +74,7 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
   let textNode: Text,
     runningOffset = nodeOffset,
     len: number;
+
```

Comment:
Call `node.normalize()` in this function. Afterwards, we shouldn't need the following `do/while` loop any more, as there will be a single TextNode. Actually, I don't think we need the `createNodeIterator` either.